frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/ol-map-manager.ts

```diff
@@ -161,19 +166,33 @@ export class OlMapManager {
             },
         });
         const viewportModify = createViewportModify(viewportLayer);
+        const simulatedRegionModify =
+            createSimulatedRegionModify(simulatedRegionLayer);
 
         const viewportTranslate = new TranslateInteraction({
             layers: [viewportLayer],
             condition: (event) => primaryAction(event) && !shiftKeyOnly(event),
             hitTolerance: 10,
         });
+        const simulatedRegionTranslate = new TranslateInteraction({
+            layers: [simulatedRegionLayer],
+            condition: (event) => primaryAction(event) && !shiftKeyOnly(event),
+            hitTolerance: 10,
+        });
 
         ModifyHelper.registerModifyEvents(viewportModify);
+        ModifyHelper.registerModifyEvents(simulatedRegionModify);
 
         const alwaysInteractions = [translateInteraction];
         const customInteractions =
             selectStateSnapshot(selectCurrentRole, this.store) === 'trainer'
-                ? [...alwaysInteractions, viewportTranslate, viewportModify]
+                ? [
+                      ...alwaysInteractions,
+                      viewportTranslate,
+                      viewportModify,
+                      simulatedRegionTranslate,
+                      simulatedRegionModify,
+                  ]
```

Comment:
You should be able to reuse the `viewportTranslate` interaction by just adding `simulatedRegionLayer` to the layers array of the translate interaction. (not tested).

Reply:
If this works, you could rename `createViewportModify()` to `createRectangularModify()`

Reply:
I like your idea, I was also thinking about that  when I saw your PR :)